xmtp_id/src/lib.rs

```diff
@@ -43,61 +58,56 @@ impl InboxOwner for LocalWallet {
     }
 }
 
-/// XMTP Identity according to [XIP-46](https://github.com/xmtp/XIPs/pull/53)
-pub struct Identity {
-    #[allow(dead_code)]
-    id: String,
-}
-
-impl Identity {
-    /// Generate a new, empty ID for an account address.
-    /// A nonce is used to ensure uniqueness of the ID.
-    pub fn new(address: String) -> Self {
-        // TODO: how to nonce?
-        let id = associations::generate_inbox_id(&address, &0);
-        Self { id }
-    }
-}
-
 #[cfg(test)]
 mod tests {
     use super::*;
     use ethers::{
-        middleware::Middleware,
+        contract::abigen,
+        middleware::SignerMiddleware,
         providers::{Http, Provider},
-        types::Address,
+        utils::Anvil,
     };
-    use std::str::FromStr;
-    use xmtp_cryptography::utils::generate_local_wallet;
+    use std::sync::Arc;
 
-    struct EthereumWallet {
-        provider: Provider<Http>,
-        address: String,
-    }
+    abigen!(
+        CoinbaseSmartWallet,
+        "artifact/CoinbaseSmartWallet.json",
+        derives(serde::Serialize, serde::Deserialize)
+    );
 
-    impl EthereumWallet {
-        pub fn new(address: String) -> Self {
-            let provider = Provider::<Http>::try_from("https://eth.llamarpc.com").unwrap();
-            Self { provider, address }
-        }
-    }
-
-    #[async_trait::async_trait]
-    impl WalletIdentity for EthereumWallet {
-        async fn is_smart_wallet(&self, block: Option<u64>) -> Result<bool, IdentityError> {
-            let address = Address::from_str(&self.address).unwrap();
-            let res = self.provider.get_code(address, block.map(Into::into)).await;
-            Ok(!res.unwrap().to_vec().is_empty())
-        }
-    }
+    abigen!(
+        CoinbaseSmartWalletFactory,
+        "artifact/CoinbaseSmartWalletFactory.json",
+        derives(serde::Serialize, serde::Deserialize)
+    );
```

Comment:
I tried to have a function that returns `(Anvil, CoinbaseSmartWalletFactory, CoinbaseSmartWallet)` but I'm not good at Generics so I copy pasted code from `erc1271_verifier.rs`.

Reply:
lets create a ticket/make it into a todo

happy to pair with you on generics at some point, too! This could be a good place to learn how to do that

Reply:
Awesome, next Monday our 1:1?